.deployment/barista/public/Jenkinsfile

```diff
@@ -0,0 +1,139 @@
+pipeline {
+  agent {
+    node {
+      label "k8s-barista-builder-public"
+    }
+  }
+
+  options {
+    ansiColor("xterm")
+    timeout(time: 4, unit: "HOURS")
+    timestamps()
+  }
+
+  environment {
+    WORKSPACE_DIR = "/home/dynatrace"
+    BARISTA_PATH = "barista"
+    ICONS_PATH = "barista-icons"
+    FRONTEND_IMAGE_NAME = "designops/barista-app-frontend"
+    BITBUCKET_URL = credentials("bitbucket-url")
+    DOCKER_REGISTRY = credentials("docker-registry")
+  }
+
+  stages {
+    stage("Prepare Workspace") {
+      steps {
+
+        sh'''
+          files="node_modules dist package-lock.sha1"
+
+          for file in $files
+          do
+            ln -s "$WORKSPACE_DIR/$file" "$PWD/$BARISTA_PATH/$file"
+          done
+        '''
+
+        dir(BARISTA_PATH) {
+
+          script {
+            def oldsha = readFile(file: "package-lock.sha1").replace('./package-lock.json', '').trim()
+            def sha1sum = sha1(file: "package-lock.json")
+
+            echo "STORED SHA: " + oldsha
+            echo "CURRENT SHA: " + sha1sum
+
+            if (!oldsha.equals(sha1sum)) {
+              echo "⚠️ Need to install packages due to updated package-lock.json"
+
+              // Delete the symlink of the node_modules first before performing an install
+              // this has to be done as ngcc has to reprocess the entries and it will fail with the symlink
+              sh '''
+                rm -rf node_modules
+                npm ci --ignore-scripts
+              '''
+            }
+          }
+
+          configFileProvider([
+            configFile(fileId: "barista-env", variable: "ENV")
+          ]) {
+            sh """
+              mv $ENV .env
+
+              echo "PUBLIC_BUILD=true" >> .env
+
+              cat .env
+            """
+          }
+        }
+      }
+    }
+
+    stage("Build Barista") {
+      steps {
+        dir(BARISTA_PATH) {
+          sh '''
+            npm run ng run barista-tools:build
+
+            if $INTERNAL_LINKS ; then
+              args='--linkReplacement=$INTERNAL_LINKS'
+            else
+              args=''
+            fi
```

Comment:
We need to replace all internal links in this builder - all the time

Reply:
Yea the environment is set so it will always use the arguments 👍